unitTests/ui/ResultListTest.ts

```diff
@@ -784,11 +784,58 @@ export function ResultListTest() {
         describe('enableScrollToTop set to', () => {
           let infiniteScrollContainer: HTMLElement;
           let allResult: IQueryResults;
+          let scrollToCalled: boolean;
+
           beforeEach(() => {
-            infiniteScrollContainer = { scrollTop: 500 } as HTMLElement;
+            scrollToCalled = false;
+            infiniteScrollContainer = ({
+              scrollTop: 500,
+              scrollTo: (x: number, y: number) => {
+                scrollToCalled = true;
+                infiniteScrollContainer.scrollTop = x;
+              }
+            } as any) as HTMLElement;
             allResult = FakeResults.createFakeResults(50);
           });
 
+          it('should scroll to top on query success', done => {
+            const option: IResultListOptions = {
+              enableInfiniteScroll: true,
+              infiniteScrollContainer: infiniteScrollContainer,
+              enableScrollToTop: true
+            };
+            test = Mock.basicComponentSetup<ResultList>(ResultList, option);
+            test.cmp.currentlyDisplayedResults.push(allResult.results[1]);
+            (test.cmp.queryController.fetchMore as jasmine.Spy).and.returnValue(Promise.resolve(allResult));
+            test.cmp.displayMoreResults(50).then(() => {
+              Simulate.query(test.env);
+              setTimeout(() => {
+                expect(scrollToCalled).toBeTruthy();
+                expect(infiniteScrollContainer.scrollTop).toBe(0);
```

Comment:
This expectation is not very meaningful because we wrote the logic to do this in the mock. Instead it would be better to check the value passed to `scrollTo` which would check that `0` is the first arg, and then the addition on line 629 in the above file as the second arg (currently not being tested for):

```
window.pageYOffset + this.searchInterface.element.getBoundingClientRect().top
```